src/de.rs

```diff
@@ -348,19 +369,21 @@ impl<'de> de::Deserializer<'de> for Deserializer {
         }
     }
 
-    // Define real `i64` / `u64` deserializers that try to cast from `f64`.
-
     fn deserialize_i64<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
-        match self.as_safe_integer() {
-            Some(v) => visitor.visit_i64(v),
-            None => self.invalid_type(visitor),
+        if self.value.is_bigint() {
```

Comment:
I'd suggest to use `.dyn_into()` here instead to combine the `is_*` check and `.into()` cast (which does its own check internally) into one.